lib/pure/os.nim

```diff
@@ -3310,3 +3310,61 @@ func isValidFilename*(filename: string, maxLen = 259.Positive): bool {.since: (1
     find(f.name, invalidFilenameChars) != -1): return false
   for invalid in invalidFilenames:
     if cmpIgnoreCase(f.name, invalid) == 0: return false
+
+when defined(windows):
+  const ComputerNamePhysicalNetBIOS = 4
+  const ComputerNamePhysicalDnsHostname = 5
+
+proc getHostName*: string {.noWeirdTarget.} =
```

Comment:
add a test in tos.nim (can't do that for setHostName unfortunately) eg;
```nim
block: # getHostName
  doAssert getHostName().len > 0
```
(better than nothing)